backend/app/anti_recommenders/anti_recommender_generator.py

```diff
@@ -1,35 +1,32 @@
 from collections.abc import Generator
-from app.utils.config.config import config
+from app.models.settings.settings import settings
 
 from app.models.anti_recommendation.anti_recommendation import AntiRecommendation
 from app.anti_recommenders.open_ai.open_ai_normal_anti_recommender.open_ai_normal_anti_recommender import (
     OpenAiNormalAntiRecommender,
 )
-from typing import TYPE_CHECKING
 
-if TYPE_CHECKING:
-    from app.anti_recommenders.anti_recommender import AntiRecommender
 
+from app.anti_recommenders.anti_recommender import AntiRecommender
 
-class AntiRecommenderProxy:
+
+class AntiRecommendationGenerator(AntiRecommender):
```

Comment:
Odd naming. Usually you'd want it to be something like `ProxyAntiRecommender`, so the name of the parent class is a suffix.

Reply:
But this is going away anyway, so don't need to rename.